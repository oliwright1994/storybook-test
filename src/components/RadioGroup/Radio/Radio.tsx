import { IRadio as IChakraIRadio, Radio as ChakraRadio } from '@chakra-ui/core'
import React, { FC } from 'react'

interface IRadio extends Pick<IChakraIRadio, 'value' | 'isDisabled' | 'isChecked' | 'onChange'> {
  id: string
}

const Radio: FC<IRadio> = ({ id, isDisabled, value, children, isChecked, onChange }) => {
  return (
    <ChakraRadio id={id} isDisabled={isDisabled} value={value} isChecked={isChecked} onChange={onChange}>
      {children}
    </ChakraRadio>
  )
}

export default Radio
