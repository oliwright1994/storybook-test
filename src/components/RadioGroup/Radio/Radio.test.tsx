import React from 'react'
import { renderWithBrightTheme } from 'testing/helpers/themeHelper'

import Radio from '.'

describe('<Radio />', () => {
  const defaultProps = {
    id: 'testRadioId',
    value: 'testRadioValue',
  }
  const label = 'Radio Button text'

  it('should render as expected', () => {
    const { container } = renderWithBrightTheme(<Radio {...defaultProps}>{label}</Radio>)

    expect(container).toMatchSnapshot()
  })

  it('should render correct text', () => {
    const { container } = renderWithBrightTheme(<Radio {...defaultProps}>{label}</Radio>)

    expect(container).toHaveTextContent(label)
  })

  it('should have disabled attribute when passed isDisabled prop', () => {
    const { container } = renderWithBrightTheme(
      <Radio {...defaultProps} isDisabled>
        {label}
      </Radio>
    )

    expect(container.querySelector(`#${defaultProps.id}`)).toBeDisabled()
  })
})
