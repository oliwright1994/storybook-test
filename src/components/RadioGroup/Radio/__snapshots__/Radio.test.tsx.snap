// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> should render as expected 1`] = `
<div>
  <label
    class="css-jwmyb8"
    for="testRadioId"
  >
    <input
      class="css-1gy22ay"
      id="testRadioId"
      type="radio"
      value="testRadioValue"
    />
    <div
      aria-hidden="true"
      class="css-1e2c42x"
      type="radio"
    >
      <span
        class="css-z7lnx4"
      />
    </div>
    <div
      class="css-6qsuox"
    >
      Radio Button text
    </div>
  </label>
</div>
`;
